--- a/cheevd_2stage.f
+++ b/cheevd_2stage.f
@@ -338,7 +338,7 @@
                LIWMIN = 1
             END IF
          END IF
-         WORK( 1 )  = CMPLX( LWMIN )
+         WORK( 1 ) = LWMIN 
          RWORK( 1 ) = REAL( LRWMIN )
          IWORK( 1 ) = LIWMIN
 *
@@ -437,7 +437,7 @@
          CALL SSCAL( IMAX, ONE / SIGMA, W, 1 )
       END IF
 *
-      WORK( 1 )  = CMPLX( LWMIN )
+      WORK( 1 ) = LWMIN 
       RWORK( 1 ) = REAL( LRWMIN )
       IWORK( 1 ) = LIWMIN
 *
