--- a/clahef.f
+++ b/clahef.f
@@ -580,7 +580,7 @@
             J = J + 1
             IF( JP.NE.JJ .AND. J.LE.N )
      $         CALL CSWAP( N-J+1, A( JP, J ), LDA, A( JJ, J ), LDA )
-         IF( J.LE.N )
+         IF( J.LT.N )
      $      GO TO 60
 *
 *        Set KB to the number of columns factorized
@@ -927,7 +927,7 @@
             J = J - 1
             IF( JP.NE.JJ .AND. J.GE.1 )
      $         CALL CSWAP( J, A( JP, 1 ), LDA, A( JJ, 1 ), LDA )
-         IF( J.GE.1 )
+         IF( J.GT.1 )
      $      GO TO 120
 *
 *        Set KB to the number of columns factorized
