--- a/zrscl.f
+++ b/zrscl.f
@@ -1,4 +1,4 @@
-*> \brief \b ZDRSCL multiplies a vector by the reciprocal of a real scalar.
+*> \brief \b ZRSCL multiplies a vector by the reciprocal of a complex scalar.
 *
 *  =========== DOCUMENTATION ===========
 *
@@ -6,7 +6,7 @@
 *            http://www.netlib.org/lapack/explore-html/
 *
 *> \htmlonly
-*> Download ZDRSCL + dependencies
+*> Download ZRSCL + dependencies
 *> <a href="http://www.netlib.org/cgi-bin/netlibfiles.tgz?format=tgz&filename=/lapack/lapack_routine/zdrscl.f">
 *> [TGZ]</a>
 *> <a href="http://www.netlib.org/cgi-bin/netlibfiles.zip?format=zip&filename=/lapack/lapack_routine/zdrscl.f">
@@ -109,7 +109,7 @@
       EXTERNAL           DLAMCH, ZLADIV
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DSCAL, ZDSCAL, ZDRSCL
+      EXTERNAL           DSCAL, ZDSCAL, ZDRSCL, ZSCAL
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS
