--- a/dorbdb4.f
+++ b/dorbdb4.f
@@ -238,7 +238,7 @@
       LOGICAL            LQUERY
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLARF, DLARFGP, DORBDB5, DROT, DSCAL,
+      EXTERNAL           DLARF1F, DLARFGP, DORBDB5, DROT, DSCAL,
      $                   XERBLA
 *     ..
 *     .. External Functions ..
@@ -279,7 +279,7 @@
          LWORKMIN = LWORKOPT
          WORK(1) = LWORKOPT
          IF( LWORK .LT. LWORKMIN .AND. .NOT.LQUERY ) THEN
-           INFO = -14
+           INFO = -15
          END IF
       END IF
       IF( INFO .NE. 0 ) THEN
