--- a/zgelss.f
+++ b/zgelss.f
@@ -215,7 +215,7 @@
       EXTERNAL           DLASCL, DLASET, XERBLA, ZBDSQR, ZCOPY,
      $                   ZDRSCL,
      $                   ZGEBRD, ZGELQF, ZGEMM, ZGEMV, ZGEQRF, ZLACPY,
-     $                   ZLASCL, ZLASET, ZUNGBR, ZUNMBR, ZUNMLQ
+     $                   ZLASCL, ZLASET, ZUNGBR, ZUNMBR, ZUNMLQ, ZUNMQR
 *     ..
 *     .. External Functions ..
       INTEGER            ILAENV
