--- a/pzpttrs.f
+++ b/pzpttrs.f
@@ -495,7 +495,7 @@
       END IF
 *
       IF( LWORK .LT. -1) THEN
-         INFO = -13
+         INFO = -14
       ELSE IF ( LWORK .EQ. -1 ) THEN
          IDUM3 = -1
       ELSE
@@ -558,7 +558,7 @@
 *
       IF( LWORK .LT. WORK_SIZE_MIN ) THEN
          IF( LWORK .NE. -1 ) THEN
-         INFO = -13
+         INFO = -14
          CALL PXERBLA( ICTXT,
      $      'PZPTTRS: worksize error',
      $      -INFO )
